.github/workflows/e2e.yaml

```diff
@@ -80,7 +80,7 @@ jobs:
         # kubectl
         curl -L https://dl.k8s.io/${{ env.K8S_VERSION }}/bin/linux/amd64/kubectl -o ${TMP_DIR}/kubectl
         # kind
-        curl -Lo ${TMP_DIR}/kind https://kind.sigs.k8s.io/dl/v0.17.0/kind-linux-amd64
+        curl -Lo ${TMP_DIR}/kind https://kind.sigs.k8s.io/dl/v0.20.0/kind-linux-amd64
```

Comment:
we will get the error if run with kind 0.17.0,  seems like it is https://github.com/kubernetes-sigs/kind/issues/3223

```
  STEP: Destroying namespace "statefulset-7177" for this suite. @ 01/17/24 08:46:32.114
  << Timeline

  [FAILED] RunHostCmd still failed after 5m3.82853998s: error running /usr/local/bin/kubectl --server=https://[::1]:34185 --kubeconfig=/home/runner/work/apiserver-network-proxy/apiserver-network-proxy/_artifacts/kubeconfig.conf --namespace=statefulset-7177 exec ss2-1 -- /bin/sh -x -c mv -v /usr/local/apache2/htdocs/index.html /tmp/ || true:
  Command stdout:

  stderr:
  error: Internal error occurred: error executing command in container: failed to exec in container: failed to start exec "f944756d1ae78400183b73440818e2b483fc255e64aa329e446e1aec455f5be9": OCI runtime exec failed: exec failed: unable to start container process: error adding pid 39487 to cgroups: failed to write 39487: openat2 /sys/fs/cgroup/unified/kubelet.slice/kubelet-kubepods.slice/kubelet-kubepods-besteffort.slice/kubelet-kubepods-besteffort-podb3dfd384_b40c_4151_aa21_aa6d091d86e4.slice/cri-containerd-e85ef49011f794f5fc1131d5e782fc37bb503a37e925a6fc93556bda5aecebbd.scope/cgroup.procs: no such file or directory: unknown
```

Do we have to use kind 0.17.0?  @aojea